controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +249,89 @@ public void invalidateCache(String table, String key) {
+    }
+
+    private CompletableFuture<List<TableSegmentKeyVersion>> removeValuesTables(String tableName, String removeValue, Set<String> tableKeys, long requestId) {
+        // Get unversioned keys.
+        final List<TableSegmentKey> keys = tableKeys.stream()
+                                                    .map(k -> TableSegmentKey.unversioned(k.getBytes(StandardCharsets.UTF_8)))
+                                                    .collect(Collectors.toList());
+
+        CompletableFuture<List<TableSegmentEntry>> f1 = segmentHelper.readTable(tableName, keys, authToken.get(), requestId) // read the latest version of keys.
+                                                                     .whenComplete((v, ex) -> releaseKeys(keys));
+        f1.thenCompose(currentEntries -> {
+            List<TableSegmentEntry> updatedList = currentEntries.stream().map(entry -> {
+                String k = entry.getKey().getKey().toString(Charset.defaultCharset());
+                byte[] array = getArray(entry.getValue());
+                byte[] updatedBytes;
+                if (array.length == 0) {
+                    updatedBytes = TagRecord.builder().tagName(k).build().toBytes();
+                } else {
+                    TagRecord record = TagRecord.fromBytes(array);
+                    TagRecord tep = record.toBuilder().removeStream(removeValue).build();
+                    updatedBytes = tep.toBytes();
+                }
+                return TableSegmentEntry.versioned(k.getBytes(StandardCharsets.UTF_8),
+                                                   updatedBytes,
+                                                   entry.getKey().getVersion().getSegmentVersion());
+
+            }).collect(Collectors.toList());
+            return segmentHelper.updateTableEntries(tableName, updatedList, authToken.get(), requestId)
+                                .whenComplete((v, ex) -> releaseEntries(updatedList));
+        });
+        return CompletableFuture.completedFuture(Collections.emptyList());
+    }
+
+    private CompletableFuture<List<TableSegmentKeyVersion>> appendValuesTable(String tableName, String appendValue, Set<String> tableKeys, long requestId) {
+        // Get unversioned keys.
+        final List<TableSegmentKey> keys = tableKeys.stream()
+                                                    .map(k -> TableSegmentKey.unversioned(k.getBytes(StandardCharsets.UTF_8)))
+                                                    .collect(Collectors.toList());
+        return segmentHelper.readTable(tableName, keys, authToken.get(), requestId) // read the latest version of keys.
+                            .whenComplete((v, ex) -> releaseKeys(keys))
+                            .thenCompose(currentEntries -> {
+
+                                List<TableSegmentEntry> updatedList = currentEntries.stream().map(entry -> {
+                                    String k = entry.getKey().getKey().toString(Charset.defaultCharset());
```

Comment:
You use `defaultCharset` here yet you use `UTF_8` above. Default Charset may differ from machine to machine so the cluster data may not be portable in case we want to failover the cluster to another location. Please use a hardcoded encoding (UTF_8 sounds good) instead of relying on system-wide defaults.

Reply:
fixed it.